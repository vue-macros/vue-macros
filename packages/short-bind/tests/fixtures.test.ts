import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import ShortBind from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.{vue,[jt]s?(x)}',
    (args, id) =>
      rollupBuild(id, [ShortBind(), RollupVue(), UnpluginOxc.rollup()]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
