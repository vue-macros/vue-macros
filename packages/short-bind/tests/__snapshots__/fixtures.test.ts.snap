// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent, ref, createElementBlock, openBlock, Fragment, createElementVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props, { expose: __expose }) {
		__expose();
		const value = ref("");
		const modelValue = ref("");
		const __returned__ = {
			value,
			modelValue
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

const _hoisted_1 = ["value"];
const _hoisted_2 = ["modelValue"];
const _hoisted_3 = ["model-value"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    Fragment,
    null,
    [
      createElementVNode("input", { value: $setup.value }, null, 8, _hoisted_1),
      createElementVNode("input", { modelValue: $setup.modelValue }, null, 8, _hoisted_2),
      createElementVNode("input", { "model-value": $setup.modelValue }, null, 8, _hoisted_3)
    ],
    64
    /* STABLE_FRAGMENT */
  );
}
var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { basic as default };
"
`;
